dcos/httpclient.go

```diff
@@ -0,0 +1,69 @@
+package dcos
+
+import (
+	"net"
+	"net/http"
+	"time"
+)
+
+const (
+	userAgent = ClientName + " - " + Version
+)
+
+type DCOSTransport struct {
+	Config *Config
+	Base   http.RoundTripper
+}
+
+func (t *DCOSTransport) base() http.RoundTripper {
+	if t.Base != nil {
+		return t.Base
+	}
+	return http.DefaultTransport
+}
+
+func (t *DCOSTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	res, err := t.base().RoundTrip(req)
+
+	req.Header.Set("User-Agent", userAgent)
```

Comment:
Hmmm, maybe I'm missing something here, but doesn't the first line in this method make the request and this just sets the header _after_ receiving a response?

Reply:
correct... was my last addition and probably being not concentrated anymore. However as told before lets handle this where the request gets formed